src/languages/elixir.js

```diff
@@ -10,111 +10,142 @@ Website: https://elixir-lang.org
     relevance: 0
   };
+  // TODO: could be tightened
+  // https://elixir-lang.readthedocs.io/en/latest/intro/18.html
+  // but you also need to include closing delemeters in the escape list per
+  // individual sigil mode from what I can tell,
+  // ie: \} might or might not be an escape depending on the sigil used
+  const ESCAPES_RE = /\\[\s\S]/;
+  // const ESCAPES_RE = /\\["'\\abdefnrstv0]/;
+  const BACKSLASH_ESCAPE = {
+    match: ESCAPES_RE,
+    scope: "char.escape",
+    relevance: 0
+  };
   const SIGIL_DELIMITERS = '[/|([{<"\']';
+  const SIGIL_DELIMITER_MODES = [
+    {
+      begin: /"/,
+      end: /"/
+    },
+    {
+      begin: /'/,
+      end: /'/
+    },
+    {
+      begin: /\//,
+      end: /\//
+    },
+    {
+      begin: /\|/,
+      end: /\|/
+    },
+    {
+      begin: /\(/,
+      end: /\)/
+    },
+    {
+      begin: /\[/,
+      end: /\]/
+    },
+    {
+      begin: /\{/,
+      end: /\}/
+    },
+    {
+      begin: /</,
+      end: />/
+    }
+  ];
   const LOWERCASE_SIGIL = {
     className: 'string',
     begin: '~[a-z]' + '(?=' + SIGIL_DELIMITERS + ')',
-    contains: [
+    contains: SIGIL_DELIMITER_MODES.map(x => hljs.inherit(x,
       {
-        endsParent: true,
         contains: [
-          {
-            contains: [
-              hljs.BACKSLASH_ESCAPE,
-              SUBST
-            ],
-            variants: [
-              {
-                begin: /"/,
-                end: /"/
-              },
-              {
-                begin: /'/,
-                end: /'/
-              },
-              {
-                begin: /\//,
-                end: /\//
-              },
-              {
-                begin: /\|/,
-                end: /\|/
-              },
-              {
-                begin: /\(/,
-                end: /\)/
-              },
-              {
-                begin: /\[/,
-                end: /\]/
-              },
-              {
-                begin: /\{/,
-                end: /\}/
-              },
-              {
-                begin: /</,
-                end: />/
-              }
-            ]
-          }
+          BACKSLASH_ESCAPE,
+          SUBST
         ]
       }
-    ]
+    ))
   };
 
   const UPCASE_SIGIL = {
     className: 'string',
     begin: '~[A-Z]' + '(?=' + SIGIL_DELIMITERS + ')',
-    contains: [
-      {
-        begin: /"/,
-        end: /"/
-      },
-      {
-        begin: /'/,
-        end: /'/
-      },
-      {
-        begin: /\//,
-        end: /\//
-      },
-      {
-        begin: /\|/,
-        end: /\|/
-      },
-      {
-        begin: /\(/,
-        end: /\)/
-      },
-      {
-        begin: /\[/,
-        end: /\]/
-      },
+    contains: SIGIL_DELIMITER_MODES.map(x => hljs.inherit(x))
+  };
+
+  const REGEX_SIGIL = {
+    className: 'regex',
+    variants: [
       {
-        begin: /\{/,
-        end: /\}/
+        begin: '~r' + '(?=' + SIGIL_DELIMITERS + ')',
```

Comment:
Will this also mark regex modifiers as part of the regex? E.g.
```
~r(hello)
~r(hello)i
~r(hello)ui
~r(hello)U
```

Reply:
No, but we should add that.  What are all the valid modifiers in Elixir?

Reply:
for sigils `r`/`R`, combinations of those letters:

```
uismxfU
```

for sigils: `w`/`W`, combinations of those letters:
```
sac
```

Treating any group of lower and/or uppercase letters immediately after the closing delimiter as a modifier and thus part of the sigil could also be enough.

Reply:
I added modifiers to ~r/~R.  I'll save `w` for another day/PR since it's not already broken out.  Really if we're going to customize these any further we perhaps need a simple data table and then programmatically auto-generate all the sigil rules from that table.  Are R and W really the only ones with modifiers?

Reply:
> Are R and W really the only ones with modifiers?

At the moment yes, but the language syntax already allows for any sigil to have any modifier that it wants. That actually makes me realize that anyone can define their own custom sigil with their own modifiers 😅.

Reply:
Problems for another day. :)  Does the PR look workable for now you think?

Reply:
I find it hard to judge the code, but from playing around with `tools/developer.html` I discovered that there is a problem with uppercase R sigil. It still needs to support escaping the closing delimiter.

For example for those two pairs the output HTML should literally only differ by a single letter (`r` <-> `R`), but right now it tries to end the R sigil too early:

```
Regex.match?(~r|foo\|bar|, "foo")
Regex.match?(~R|foo\|bar|, "foo")


Regex.match?(~r(hello( there\)*!)u, "hello!")
Regex.match?(~R(hello( there\)*!)u, "hello!")
```

Reply:
Because I thought uppercase sigils didn't allow escaping.  I guess escaping the end sigil character is the exception to the rule?

Reply:
Yes, that is the exception. Quoting the docs about the sigil R:

> It returns a regular expression pattern without interpolations and without escape characters. Note it still supports escape of Regex tokens (such as escaping + or ?) and it also requires you to escape the closing sigil character itself if it appears on the Regex.